core/src/box_kind/pool_box.rs

```diff
@@ -90,6 +112,38 @@ impl TryFrom<ErgoBox> for PoolBoxWrapper {
             return Err(PoolBoxError::NoEpochCounter);
         }
 
-        Ok(Self(b))
+        let reward_token_id = b
+            .tokens
+            .as_ref()
+            .ok_or(PoolBoxError::NoTokens)?
+            .get(1)
+            .ok_or(PoolBoxError::NoRewardToken)?
+            .token_id
+            .clone();
+        if reward_token_id
```

Comment:
Sorry, I forgot this one after hitting 'submit review':

I think we shouldn't be comparing a hard-coded reward token id here, but rather pulling the value from `OracleConfig::reward_token_id`?

I did a grep, and saw a similar thing here: https://github.com/ergoplatform/oracle-core/blob/aa0d154670677984e1024a5dde1bc75daf43b5f3/core/src/box_kind/oracle_box.rs#L65-L66

Reply:
Good catch! I had a task about that among my todos but then deleted it for whatever reason. Let's clean this up.
All those hardcoded reward token ids are in tests. Since we don't have reward token id hardcoded in the contracts I'd rather use random ids in the tests.